api/libproject/devstart.rb

```diff
@@ -370,10 +370,26 @@ def run_public_api_tests(cmd_name, args)
   :fn => ->(*args) { run_public_api_tests("test-public-api", args) }
 })
 
+def test_api_changes(branch_name)
+  common = Common.new
+  common_commit = common.capture_stdout(%W{git merge-base master blrubenstein/test-circle}).split("\n")[0]
+  api_lines = common.capture_stdout(%W{git diff --name-only #{common_commit}}).split("\n")
+  return api_lines.any? { |s| s.include?('api') }
+end
 
 def run_all_tests(cmd_name, args)
-  run_api_tests(cmd_name, args)
-  run_public_api_tests(cmd_name, args)
+  common = Common.new
+  circle_branch = ENV["CIRCLE_BRANCH"]
```

Comment:
I still feel it would be cleaner to move this logic into a circle-specific shell script and invoke it from the circle config. Ideally project.rb would not be aware that it's running within Circle or even git.